feature_engine/selection/drop_correlated_features.py

```diff
@@ -186,6 +212,14 @@ def fit(self, X: pd.DataFrame, y: pd.Series = None):
         # create tuples of correlated feature groups
         self.correlated_feature_sets_ = []
 
+        # sort columns for consistency
+        if self.order_by == "nan":
+            X = X[X.isna().sum(0).sort_values().index]
+        elif self.order_by == "unique":
+            X = X[X.nunique(0).sort_values(ascending=False).index]
+        elif self.order_by == "alphabetic":
+            X = X.sort_index(axis=1)
```

Comment:
I think it'd be good to have this in a bespoke hidden method `(_sort_variables(self)`, so that we can test just the ordering functionality without having to go through the whole fit transform process to get something useful.